apis/v1alpha2/validation/validation_test.go

```diff
@@ -424,3 +426,76 @@ func portNumberPtr(p int) *gatewayv1a2.PortNumber {
 	result := gatewayv1a2.PortNumber(p)
 	return &result
 }
+
+func TestValidateGatewayClassUpdate(t *testing.T) {
+	type args struct {
+		oldClass *gatewayv1a2.GatewayClass
+		newClass *gatewayv1a2.GatewayClass
+	}
+	tests := []struct {
+		name string
+		args args
+		want field.ErrorList
+	}{
+		{
+			name: "changing parameters reference is allowed",
+			args: args{
+				oldClass: &gatewayv1a2.GatewayClass{
+					Spec: gatewayv1a2.GatewayClassSpec{
+						Controller: "foo",
+					},
+				},
+				newClass: &gatewayv1a2.GatewayClass{
+					Spec: gatewayv1a2.GatewayClassSpec{
+						Controller: "foo",
+						ParametersRef: &gatewayv1a2.ParametersReference{
+							Group: "example.com",
+							Kind:  "GatewayClassConfig",
+							Name:  "foo",
+						},
+					},
+				},
+			},
+			want: nil,
+		},
+		{
+			name: "changing controller field results in an error",
+			args: args{
+				oldClass: &gatewayv1a2.GatewayClass{
+					Spec: gatewayv1a2.GatewayClassSpec{
+						Controller: "foo",
+					},
+				},
+				newClass: &gatewayv1a2.GatewayClass{
+					Spec: gatewayv1a2.GatewayClassSpec{
+						Controller: "bar",
+					},
+				},
+			},
+			want: field.ErrorList{
+				{
+					Type:     field.ErrorTypeInvalid,
```

Comment:
my preference in these kind of tests is to just test for the error Type and not do DeepEqual. DeepEqual makes it so that we have to change a lot of tests if the error message needs to be updated.

Reply:
Can a smart replace tool not help us there?
I'm not a huge fan of the current approach but it helps when there are tests that have multiple errors of the same type.

Reply:
Yeah in this case I think it's useful to check for the error message since that is an important part of the output here.  If it gets more complex in the future we may need to revisit this, but for now I think it's good for the tests to cover the message.